app/coffeescript/components/data/info_window.coffee

```diff
@@ -3,22 +3,42 @@
 define [
   'jquery'
   'flight/lib/component'
-  'map/components/mixins/map_utils'
+  'underscore'
 ], (
   $
   defineComponent
-  mapUtils
+  _
 ) ->
 
   infoWindowData = ->
 
     @defaultAttrs
-      pinRoute: "/map/pin/"
+      route: "/map/pin/"
+      allowed_filters: ['min_price', 'max_price']
+      refinements: {}
 
+    @formatRefinements = (results) ->
+      return "" if _.isEmpty(results.names)
+      "refinements=#{results.names.join('-')}-#{results.ids.join('+')}"
+
+    @formatFilters = (filters, names) ->
+      return "" if _.isEmpty(filters)
+      if _.isEmpty(names) then filters.join('&') else "&#{filters.join('&')}"
+
+    @queryParams = ->
+      return "" if _.isEmpty(@attr.refinements)
+      results = {names: [], ids: [], filters: []}
+      for own name, obj of @attr.refinements
+        if _.contains(@attr.allowed_filters, name)
+          results.filters.push "#{obj.dim_id}=#{obj.value}"
+        else
+          results.names.push obj.dim_name
+          results.ids.push obj.dim_id
+      "?#{@formatRefinements(results)}#{@formatFilters(results.filters, results.names)}"
```

Comment:
Can we separate the logic from building the url string? For example, use this function to build a generic `results` and other to take an object and turn it into a query string.